test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusIntegrationTestExtension.java

```diff
@@ -146,8 +146,14 @@ private QuarkusTestExtensionState ensureStarted(ExtensionContext extensionContex
         QuarkusTestExtensionState state = getState(extensionContext);
         Class<? extends QuarkusTestProfile> selectedProfile = findProfile(testClass);
         boolean wrongProfile = !Objects.equals(selectedProfile, quarkusTestProfile);
+        // we reset the failed state if we changed test class
+        boolean resetFailedState = !Objects.equals(extensionContext.getRequiredTestClass(), currentJUnitTestClass);
+        if (resetFailedState && state != null) {
+            state.setTestFailed(null);
+            currentJUnitTestClass = extensionContext.getRequiredTestClass();
+        }
         // we reload the test resources if we changed test class and if we had or will have per-test test resources
-        boolean reloadTestResources = !Objects.equals(extensionContext.getRequiredTestClass(), currentJUnitTestClass)
+        boolean reloadTestResources = resetFailedState
```

Comment:
This is kind of deceiving in appearances TBH because whether or not `reloadTestResources` is set to true, is not theorically equivalent to setting `resetFailedState` to `true`. 

So I would just keep this line the want it was.

Reply:
If you want to avoid code duplication what I would do is introduce 👍🏼 

```java
boolean isNewTestClass = !Objects.equals(extensionContext.getRequiredTestClass(), currentJUnitTestClass);
        if (resetFailedState && state != null) {
```

and use that to set `resetFailedState` and `reloadTestResources`

Reply:
Good point. Done.